### PR TITLE
LPS-184647,LPS-184643,LPS-184644

### PR DESCRIPTION
**Pull request by ces-totrinh**
Hi team,
Please help review this PR and leave your comments.
Previous PR: https://github.com/tototrinh/liferay-portal/pull/66
# Motivation
[LPS-184647 ](https://liferay.atlassian.net/browse/LPS-184647) [BE] Add the sub-folder permission base on the parent folder setting ( permission propagation is set to true or false)
[LPS-184643 ](https://liferay.atlassian.net/browse/LPS-184643) [FE] Add a message to show that permission of the sub-folder will be inherited from the parent folder when permission propagation of the parent folder is set to true in Creating new sub-folder view
[LPS-184644 ](https://liferay.atlassian.net/browse/LPS-184644) [FE] Show the default Permission setting when the parent folder with permission propagation is set to false in creating new sub-folder view

Confirmation from the Prod Team for which parent folder, the new folder will inherit permission from:
LPS-154515
# Solution
Create logic to determine the folderId of the inheritable parent folder and then use it to propagate the permissions to the new folder.
Thanks.
cc @ces-quanhuynh  @ces-luannguyen @locpham97

### CODE REVIEW

**Comment by ces-quanhuynh** [comment]
From my perspective, we should squash the first two commits. Because the "Inherit permissions with correct parent folder" commit changed some logic and made it clearer for inheriting roles permissions. Moreover, it can help Product team have a clear observation of the logic.

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLEditFolderDisplayContext.java [hunk 1]
```diff
@@ -150,6 +152,55 @@ public String getHeaderTitle() {
 		return LanguageUtil.get(_httpServletRequest, "new-folder");
 	}
 
+	public long getInheritableParentFolderId() {
+		List<Long> ancestorIds = new ArrayList<>();
+
+		Folder folder = _getParentFolder();
```
Comment (ces-quanhuynh) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLEditFolderDisplayContext.java:158]: I think this should be in the try block

Comment (tototrinh) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLEditFolderDisplayContext.java]: I updated the code. Thanks.

**Review by ces-quanhuynh — APPROVED** [review]
LGTM!

File: portal-impl/src/com/liferay/portlet/documentlibrary/util/packageinfo [hunk 2]
```diff
@@ -1 +1 @@
-version 8.1.0
\ No newline at end of file
+version 8.2.0
```
Comment (locpham97) [inline portal-impl/src/com/liferay/portlet/documentlibrary/util/packageinfo:1]: baseline it

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_folder.jsp [hunk 3]
```diff
@@ -215,11 +215,22 @@ renderResponse.setTitle(dlEditFolderDisplayContext.getHeaderTitle());
...
 						<aui:fieldset collapsed="<%= true %>" collapsible="<%= true %>" label="permissions">
-							<liferay-ui:input-permissions
-								modelName="<%= DLFolderConstants.getClassName() %>"
-								permissionPropagationCheckboxLabel="set-permissions-for-documents-and-folders-uploaded-to-this-folder"
-								permissionPropagationCheckboxValue="<%= true %>"
-							/>
+							<c:choose>
+								<c:when test="<%= dlEditFolderDisplayContext.getInheritableParentFolderId() >= 0 %>">
+									<label class="font-weight-normal mb-3">
+										<liferay-ui:message arguments='<%= "<strong>" + HtmlUtil.escape(dlEditFolderDisplayContext.getInheritableParentFolderName()) + "</strong>" %>' key="these-permissions-are-inherited-from-the-parent-folder" />
```
Comment (locpham97) [inline modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_folder.jsp:221]: Not sure. However, I think it is better to change  -> getInheritedParentFolderName

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLEditFolderDisplayContext.java [hunk 4]
```diff
@@ -150,6 +152,55 @@ public String getHeaderTitle() {
...
+			folder = DLAppServiceUtil.getFolder(inheritableParentFolderId);
+		}
+		catch (PortalException portalException) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(portalException);
+			}
+		}
+
+		if (folder != null) {
+			return folder.getName();
```
Comment (locpham97) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLEditFolderDisplayContext.java:198]: better to move to try

File: portal-kernel/src/com/liferay/document/library/kernel/model/DLFolderConstants.java [hunk 5]
```diff
@@ -24,6 +24,8 @@
  */
 public class DLFolderConstants {
 
+	public static final boolean DEFAULT_FOLDER_PERMISSION_PROPAGATION = true;
```
Comment (locpham97) [inline portal-kernel/src/com/liferay/document/library/kernel/model/DLFolderConstants.java:27]: ENABLED :3 maybe good

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_folder.jsp (hunk 3 above)
Comment (tototrinh) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_folder.jsp:221]: `Inheritable` means `capable of being inherited` and I think it is more correct in this case than `Inherited`. Thanks.

**Review by ces-locpham — APPROVED** [review]

**Comment by ces-totrinh** [comment]
@hhtank this RP passed our internal review. Please help take a look. Thanks.

**Comment by ces-totrinh** [comment]
@locpham97 @ces-quanhuynh
I have updated the code. When creating the folder, if the Home folder permissions do not exist, initialize the permissions of the Home folder and then allow the newly created folder to inherit the permissions.
@hhtank I found the issue and updated the fix. This PR needs to be reconsidered so I will let you know when the internal review is finished. Sorry for the inconvenience.

**Review by ces-locpham — APPROVED** [review]

**Review by ces-quanhuynh — APPROVED** [review]

**Comment by ces-totrinh** [comment]
Hi @hhtank This PR passed our internal review. Please help review it and leave your feedback. Many thanks.

**Comment by hhtank** [comment]
@tototrinh LGTM, thanks.

**Comment by ces-totrinh** [comment]
Merged. Thanks all.

**Pull request merged**